Tests/Integration/LnsProtocolMessageProcessorConcentratorDeduplicationIntegrationTests.cs

```diff
@@ -108,5 +108,11 @@ public async Task When_Same_Message_Comes_Multiple_Times_Result_Depends_On_Which
             var expected = isSameStation ? 2 : 1;
             Assert.Equal(expected, dispatcherCounter);
         }
+
+        public void Dispose()
+        {
+            _ = this.testHost?.StopAsync();
```

Comment:
Since this is not awaited, I can't imagine how this is working unless `StopAsync` is (incidentally) always completing synchronously before being disposed. I believe the right fix here is to implement `IAsyncLifetime` and stop with an await in `DisposeAsync`.

Reply:
🤦‍♂️I should have known better here, thanks for the catch.

Reply:
Btw not sure why R# complains here, not sure how a NullPointerReference can even occur here
```
var obj = new A();
await obj?.DisposeAsync();

class A
{
	public async Task DisposeAsync()
	{
		await Task.Delay(1);
	}
}
```